File: ejercicios-py/ejercicio1.py
from tabulate import tabulate
from typing import List, Dict, Any
import sys
# Definir la funcion para realizar el cálculo
def calcular(num1: float, num2: float, operacion: str) -> float:
    if operacion == '+':
        return num1 + num2
    elif operacion == '-':
        return num1 - num2
    elif operacion == '*':
        return num1 * num2
    elif operacion == '/':
        return "Error: división por cero" if num2 == 0 else num1 / num2
    else:
        return "Operación no válida"

def mostrar_resultados(resultados: List[Dict[str, Any]], titulo: str) -> None:
    print(f"\n{titulo}")
    print(tabulate(resultados, headers="keys", tablefmt="grid"))
    sys.stdout.flush()  # Forzar la salida inmediata

# Datos para pruebas
numeros = [5, 3, 10, 0, 2, 4, 10, 2]
operaciones = ['+', '-', '*', '/']

print("🧮 Calculadora avanzada en Python\n")
sys.stdout.flush()

# 1. Usando for tradicional
print("Iniciando sección 1...")
resultados_for = []
for i in range(0, len(numeros), 2):
    for op in operaciones:
        if i + 1 < len(numeros):
            resultados_for.append({
                'Número 1': numeros[i],
                'Número 2': numeros[i + 1],
                'Operación': op,
                'Resultado': calcular(numeros[i], numeros[i + 1], op)
            })
mostrar_resultados(resultados_for, "1️⃣ Operaciones con for tradicional:")

print("\n✅ Ejecución completada")
sys.stdout.flush()

File: ejercicios-js/ejercicio1.js
// Funcion para calcular operaciones basicas
function calcular(num1, num2, operacion) {
    switch (operacion) {
      case '+':
        return num1 + num2;
      case '-':
        return num1 - num2;
      case '*':
        return num1 * num2;
      case '/':
        return num2 === 0 ? "Error: división por cero" : num1 / num2;
      default:
        return "Operación no válida";
    }
}

// Array de números para pruebas. Genera aleatoriamente 100 números entre 0 y 100
const numeros = Array.from({ length: 10 }, () => Math.floor(Math.random() * 100));
const operaciones = ['+', '-', '*', '/'];

// Función para mostrar resultados en formato tabla
function mostrarResultados(resultados) {
    console.table(resultados);
}

// Pruebas con diferentes tipos de ciclos
console.group('🧮 Calculadora Avanzada');

// 1. Usando for tradicional
console.info('1️⃣ Operaciones con for tradicional:');
const resultadosFor = [];
for (let i = 0; i < numeros.length; i += 2) {
    for (let j = 0; j < operaciones.length; j++) {
        resultadosFor.push({
            'Número 1': numeros[i],
            'Número 2': numeros[i + 1],
            'Operación': operaciones[j],
            'Resultado': calcular(numeros[i], numeros[i + 1], operaciones[j])
        });
    }
}
mostrarResultados(resultadosFor);
console.groupEnd();